src/main/java/seedu/ravi/Executor.java

```diff
@@ -19,6 +20,7 @@ public static CommandResult executeCommand(String userInput) {
 
         if (command.getPromptType() == PromptType.EDIT) {
             StateManager.saveState(userInput);
+            InputOutputManager.save();
```

Comment:
Why is the saving moved to the executor?

Reply:
I moved it here because it was stated in this guide that the shutdown hook should be fast. So I decided to move it to `PromptType.Edit` commands.

https://www.geeksforgeeks.org/jvm-shutdown-hook-java/

![image](https://user-images.githubusercontent.com/45010376/98460003-20fdf200-21db-11eb-8aa1-62fc6e8b5969.png)
![image](https://user-images.githubusercontent.com/45010376/98460004-265b3c80-21db-11eb-9e71-c20f69dfb120.png)